src/Symfony/Component/Workflow/Definition.php

```diff
@@ -17,8 +17,9 @@
 /**
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
  */
-class Definition
+final class Definition
```

Comment:
A: Hey, guys, this class is self-sufficient and we don't see sense in other implementations. Also, we want to encourage to use composition over inheritance. Let's mark it `final`.
B: Don't use `final`.
A: Why?
B: Because it makes impossible to extend this class nor replace implementation.
A: ... That's the point. You shouldn't do it.

Our dialog looks like above as long as you don't provide real use case when you _need_ to inherit this class. It's perfectly fine to have the only implementation sometimes, especially for VOs.

> Because then there is neither a way how to switch nor extend implementation.

It's like saying that poison shouldn't be marked with "don't eat it" label, because you want to eat it. Well, that's not the purpose of the object. Don't ~eat~ extend it.